cunyfirstapi/class_search.py

```diff
@@ -0,0 +1,308 @@
+                    'DERIVED_SSS_SCL_SSS_MORE_FINANCES': '9999',
+                    #'CU_SF_SS_INS_WK_BUSINESS_UNIT': 'QNS01',
+                    'DERIVED_SSS_SCL_SSS_MORE_PROFILE': '9999',
+                    'ptus_defaultlocalnode': 'PSFT_CNYHCPRD',
+                    'ptus_dbname': 'CNYHCPRD',
+                    'ptus_portal': 'EMPLOYEE',
+                    'ptus_node': 'HRMS',
+                    'ptus_workcenterid': ''}
+        
+        #pprint(self._session.cookies.get_dict())
+
+        self._response = self._session.post(url=constants.CUNY_FIRST_STUDENT_CENTER_BASE_URL, 
+            data=payload, 
+            headers=headers)
+        #print(self._response.text)
+        self._response = self._session.get(url=constants.CUNY_FIRST_CLASS_SEARCH_URL, 
+            headers=headers)
+        #print(self._response.text)
+
+        return self
+    def action(self):
+        return Class_Search_Action(self)
+
+
+class Class_Search_Action(ActionObject):
+
+    def __init__(self, location):
+        self._location = location
+
+    def location(self):
+        return self._location #Class_Search()
+
+
+    def submit_search(self, institution, term, course_number, 
+                        subject='', course_number_match='E', 
+                        course_career='', course_attribute='', 
+                        course_attribute_value='', 
+                        requirement_designation='',
+                        open_classes_only=True, session='', 
+                        mode_of_instruction='', 
+                        meeting_start_time_match='GE',
+                        meeting_start_time='', 
+                        meeting_end_time_match='LE',
+                        meeting_end_time='', days_of_week_match='', 
+                        days_of_week=None, class_number='', 
+                        course_keyword='',
+                        minimum_units_match='GE', minimum_units='', 
+                        maximum_units_match='LE', maximum_units='', 
+                        course_component='', campus='', _location='', 
+                        instructor_last_name_match='B', 
+                        instructor_last_name='', parsed=True):
+        
+        if days_of_week is None:
+            days_of_week = []
+
+        days_of_week = set(map(lambda x: x.lower(), days_of_week))
+
+        headers = {
+            'Origin': 'https://hrsa.cunyfirst.cuny.edu',
+            'Accept-Encoding': 'gzip, deflate, br',
+            'Accept-Language': 'en-US,en;q=0.9',
+            'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36',
+            'Content-Type': 'application/x-www-form-urlencoded',
+            'Accept': '*/*',
+            'Referer': constants.CUNY_FIRST_CLASS_SEARCH_URL,
+            'Connection': 'keep-alive',
+        }  
+        '''  
+        ICAction: Tell the server we are gonna search for a class now
```

Comment:
This comment applies to all the docstrings below. These are written well! But. Any def of a variable (term, course career, institution ...etc) need to be placed in 1 docstring above the function declaration.

Reply:
Short 1/2 sentence blurb should go above these broken up dictionary form pieces explaining which part they are.